### PR TITLE
feat(trie): rocksdb update 7.7.3

### PR DESCRIPTION
**Pull request by halibobo1205**
**What does this PR do?**

**Why are these changes required?**

**This PR has been tested by:**
- Unit Tests
- Manual Testing

**Follow up**

**Extra details**

### CODE REVIEW

File: chainbase/src/main/java/org/tron/core/state/WorldStateGenesis.java [hunk 1]
```diff
@@ -177,13 +186,26 @@ private void initGenesisDBs() {
...
+          } catch (RocksDBException | IOException e) {
             throw new RuntimeException(e);
           }
         });
   }
 
+  private DB getDb(Path sourceDir, String dbName) throws RocksDBException, IOException {
+    File engineFile = Paths.get(sourceDir.toString(), dbName, ENGINE_FILE).toFile();
+    if (!engineFile.exists()) {
+      return  new RocksDB(sourceDir, dbName);
```
Comment (guoquanwu) [inline chainbase/src/main/java/org/tron/core/state/WorldStateGenesis.java:199]: If the file does not exist, is it a LevelDB?

File: chainbase/src/main/java/org/tron/core/state/WorldStateGenesis.java [hunk 2]
```diff
@@ -328,5 +350,74 @@ public void close() throws IOException {
...
+
+    private static final String MARKET_PAIR_PRICE_TO_ORDER = "market_pair_price_to_order";
+
+    public LevelDB(Path path, String name) throws IOException {
+      this.name = name;
+      this.levelDB = newLevelDb(Paths.get(path.toString(), name));
+    }
+
+    @Override
+    public byte[] get(byte[] key) throws RocksDBException {
```
Comment (guoquanwu) [inline chainbase/src/main/java/org/tron/core/state/WorldStateGenesis.java:366]: may not a RocksDBException?

File: chainbase/src/main/java/org/tron/core/state/WorldStateGenesis.java (hunk 1 above)
Comment (halibobo1205) [reply chainbase/src/main/java/org/tron/core/state/WorldStateGenesis.java]: yes

File: chainbase/src/main/java/org/tron/core/state/WorldStateGenesis.java (hunk 2 above)
Comment (halibobo1205) [reply chainbase/src/main/java/org/tron/core/state/WorldStateGenesis.java]: yes

**Review by guoquanwu — APPROVED** [review]

**Pull request merged**